src/passes/pass.cpp

```diff
@@ -423,7 +423,11 @@ void PassRunner::addDefaultFunctionOptimizationPasses() {
   // that depend on flat IR
   if (options.optimizeLevel >= 4) {
     addIfNoDWARFIssues("flatten");
+    // LocalCSE is particularly useful after flatten, but we must simplify
+    // locals a little first (as flatten adds many new and redundant ones).
```

Comment:
This was referring to this comment in the pass itself:
```cpp
// This pass only finds entire expression trees, and not parts of them, so we
// will not optimize this:
//
//  (A (B (C (D1))))
//  (A (B (C (D2))))
//
// The innermost child is different, so the trees are not identical. However,
// running flatten before running this pass would allow those to be optimized as
// well.
```
I'll improve the comment here to refer to there, and explain why simplify-locals is needed (flatten adds new redundant locals, so things look different that are actually equal, until we simplify locals).